api/src/utils/get-challenges.ts

```diff
@@ -0,0 +1,37 @@
+import { flatten } from 'lodash';
+
+// TODO: keeping curriculum in memory is handy if we want to field requests that
+// need to 'query' the curriculum, but if we force the client to handle
+// redirectToCurrentChallenge and, instead, only report the current challenge id
+// via the user object, then we should *not* store this so it can be garbage
+// collected.
+import curriculum from '../../../config/curriculum.json';
+import { SuperBlocks } from '../../../config/superblocks';
+import { type ChallengeNode } from '../../../client/src/redux/prop-types';
+
+type Curriculum = { [keyValue in SuperBlocks]?: CurriculumProps };
+type SuperBlockKeys = keyof Curriculum;
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+interface Block<T> {
+  challenges: ChallengeNode['challenge'][];
+}
```

Comment:
```suggestion
interface Block {
  challenges: {
    id: string;
    tests?: { id?: string }[];
    challengeType: number;
  }[];
}
```
This is probably good enough for now, since we're likely refactoring all of this soon.